response.go

```diff
@@ -215,253 +215,378 @@ func visitModelNode(model interface{}, included *map[string]*Node,
+			return node
+		}
+
+		// Deal with database/sql null types
+		if nBool, ok := fieldValue.Interface().(sql.NullBool); ok {
+			if !nBool.Valid && omitEmpty {
+				return node
 			}
-		} else if annotation == annotationAttribute {
-			var omitEmpty, iso8601 bool
-
-			if len(args) > 2 {
-				for _, arg := range args[2:] {
-					switch arg {
-					case annotationOmitEmpty:
-						omitEmpty = true
-					case annotationISO8601:
-						iso8601 = true
-					}
-				}
+
+			if nBool.Valid {
+				node.Attributes[args[1]] = nBool.Bool
+			} else {
+				node.Attributes[args[1]] = nil
 			}
+			break
+		}
 
-			if node.Attributes == nil {
-				node.Attributes = make(map[string]interface{})
+		if nStr, ok := fieldValue.Interface().(sql.NullString); ok {
+			if !nStr.Valid && omitEmpty {
+				return node
 			}
 
-			if fieldValue.Type() == reflect.TypeOf(time.Time{}) {
-				t := fieldValue.Interface().(time.Time)
-
-				if t.IsZero() {
-					continue
-				}
-
-				if iso8601 {
-					node.Attributes[args[1]] = t.UTC().Format(iso8601TimeFormat)
-				} else {
-					node.Attributes[args[1]] = t.Unix()
-				}
-			} else if fieldValue.Type() == reflect.TypeOf(new(time.Time)) {
-				// A time pointer may be nil
-				if fieldValue.IsNil() {
-					if omitEmpty {
-						continue
-					}
-
-					node.Attributes[args[1]] = nil
-				} else {
-					tm := fieldValue.Interface().(*time.Time)
-
-					if tm.IsZero() && omitEmpty {
-						continue
-					}
-
-					if iso8601 {
-						node.Attributes[args[1]] = tm.UTC().Format(iso8601TimeFormat)
-					} else {
-						node.Attributes[args[1]] = tm.Unix()
-					}
-				}
+			if nStr.Valid {
+				node.Attributes[args[1]] = nStr.String
 			} else {
-				// Dealing with a fieldValue that is not a time
-				emptyValue := reflect.Zero(fieldValue.Type())
-
-				// See if we need to omit this field
-				if omitEmpty && reflect.DeepEqual(fieldValue.Interface(), emptyValue.Interface()) {
-					continue
-				}
-
-				strAttr, ok := fieldValue.Interface().(string)
-				if ok {
-					node.Attributes[args[1]] = strAttr
-				} else {
-					node.Attributes[args[1]] = fieldValue.Interface()
-				}
+				node.Attributes[args[1]] = nil
 			}
-		} else if annotation == annotationRelation {
-			var omitEmpty bool
+			break
+		}
 
-			//add support for 'omitempty' struct tag for marshaling as absent
-			if len(args) > 2 {
-				omitEmpty = args[2] == annotationOmitEmpty
+		if nF64, ok := fieldValue.Interface().(sql.NullFloat64); ok {
+			if !nF64.Valid && omitEmpty {
+				return node
 			}
 
-			isSlice := fieldValue.Type().Kind() == reflect.Slice
-			if omitEmpty &&
-				(isSlice && fieldValue.Len() < 1 ||
-					(!isSlice && fieldValue.IsNil())) {
-				continue
+			if nF64.Valid {
+				node.Attributes[args[1]] = nF64.Float64
+			} else {
+				node.Attributes[args[1]] = nil
 			}
+			break
+		}
 
-			if node.Relationships == nil {
-				node.Relationships = make(map[string]interface{})
+		if nI32, ok := fieldValue.Interface().(sql.NullInt32); ok {
```

Comment:
I couldn't find a DRY way to handle the repetitive checks `sql.Null*` types require (omit attribute, return `null` and return value).